nxdrive/metrics/poll_metrics.py

```diff
@@ -0,0 +1,69 @@
+from logging import getLogger
+from queue import Empty, Queue
+from time import monotonic_ns
+from typing import TYPE_CHECKING
+
+from ..constants import TIMEOUT
+from ..engine.workers import PollWorker
+from ..exceptions import ThreadInterrupt
+from ..objects import Metrics
+from ..options import Options
+from ..qt.imports import pyqtSlot
+from .constants import ENDPOINT, SYNC_ACTION, SYNC_TIME
+
+log = getLogger(__name__)
+
+
+if TYPE_CHECKING:
+    from ..client.remote_client import Remote
+
+_CHECK_INTERVAL = 900  # 15 minutes
+
+
+class CustomPollMetrics(PollWorker):
+    def __init__(self, remote: "Remote") -> None:
+        super().__init__(_CHECK_INTERVAL, "CustomMetrics")
+
+        self._remote = remote
+        self._metrics_queue: Queue = Queue()
+
+        self.timeout = Options.timeout if Options.timeout > 0 else TIMEOUT
+
+    def _poll(self) -> None:
+        errors = []
+        try:
+            while True:
+                try:
+                    metrics = self._metrics_queue.get_nowait()
+                    print(f"Sending {metrics}")  # DEV
+                    if not Options.custom_metrics:
+                        continue
+                    self._remote.client.request(
+                        "GET", ENDPOINT, headers=metrics, timeout=self.timeout
+                    )
+                except Empty:
+                    break
+                except Exception:
+                    log.warning(
+                        "Could not send metrics. Pushing to queue.", exc_info=True
+                    )
+                    errors.append(metrics)
+                    continue
+        except ThreadInterrupt:
+            raise
+        finally:
+            for elem in errors:
+                self.send(elem)
```

Comment:
Hm maybe something less "one-block" would suit better:
```suggestion
        while "there are metrics":
            try:
                metrics = self._metrics_queue.get_nowait()
            except Empty:
                break

            if not Options.custom_metrics:
                continue

            print(f"Sending {metrics}")  # DEV
            try:
                self._remote.client.request(
                    "GET", ENDPOINT, headers=metrics, timeout=self.timeout
                )
            except ThreadInterrupt:
                self.send(metrics)  # Repush to the queue
                raise from None
            except Exception:
                log.warning("Could not send metrics", exc_info=True)
                self.send(metrics)  # Repush to the queue
```